core/src/main/java/io/grpc/internal/DelayedStream.java

```diff
@@ -217,10 +217,8 @@ public void run() {
 
   @Override
   public Attributes getAttributes() {
-    if (passThrough) {
-      return realStream.getAttributes();
-    }
-    return Attributes.EMPTY;
+    checkState(passThrough, "passThrough");
```

Comment:
That string isn't helpful for users. How about "Called getAttributes before attributes are ready." Or similar.